src/__tests__/resources/revisions3D.integration.spec.ts

```diff
@@ -0,0 +1,175 @@
+  });
+
+  const now = Date.now();
+  let revisions: Revision3D[];
+  let file: UploadFileMetadataResponse;
+  let model: Model3D;
+
+  test('create model', async () => {
+    const modelToCreate = { name: `Model revision test ${now}` };
+    model = (await client.models3D.create([modelToCreate]))[0];
+    expect(model).toBeTruthy();
+  });
+
+  test('create file', async () => {
+    const fileContent = readFileSync('src/__tests__/test3dFile.fbx');
+    const fileProps = { name: `file_revision_test_${now}.fbx` };
+    file = await client.files.upload(fileProps, fileContent);
+    expect(file).toBeTruthy();
+  });
+
+  test(
+    'create revision',
+    async done => {
+      const revisionsToCreate: CreateRevision3D[] = [{ fileId: file.id }];
+      revisions = await retryInSeconds(
+        () => client.revisions3D.create(model.id, revisionsToCreate),
+        3,
+        400,
+        60
+      );
+      expect(revisions.length).toBe(1);
+      done();
+    },
+    10 * 1000
+  );
+
+  test('retrieve', async () => {
+    const retrieved = await client.revisions3D.retrieve(
+      model.id,
+      revisions[0].id
+    );
+    expect(retrieved.fileId).toBe(file.id);
+    expect(retrieved.published).toBeFalsy();
+    expect(retrieved.rotation).toBeFalsy();
+  });
+
+  test('update', async () => {
+    const newRotation: Tuple3<number> = [3, 14, 15];
+    const newCameraTarget: Tuple3<number> = [1, 2, 3];
+    const newCameraPosition: Tuple3<number> = [3, 2, 1];
+    const revisionsToUpdate: UpdateRevision3D[] = revisions.map(revision => ({
+      id: revision.id,
+      update: {
+        rotation: {
+          set: newRotation,
+        },
+        camera: {
+          set: {
+            target: newCameraTarget,
+            position: newCameraPosition,
+          },
+        },
+      },
+    }));
+    const updatedRevisions = await client.revisions3D.update(
+      model.id,
+      revisionsToUpdate
+    );
+    updatedRevisions.forEach(revision =>
+      expect(revision.rotation).toEqual(newRotation)
+    );
+    updatedRevisions.forEach(revision =>
+      expect(revision.camera && revision.camera.target).toEqual(newCameraTarget)
+    );
+    updatedRevisions.forEach(revision =>
+      expect(revision.camera && revision.camera.position).toEqual(
+        newCameraPosition
+      )
+    );
+  });
+
+  test('list', async () => {
+    const listed = await client.revisions3D.list(model.id).autoPagingToArray();
+    expect(revisions.map(r => r.id).sort(simpleCompare)).toEqual(
+      listed.map(r => r.id).sort(simpleCompare)
+    );
+  });
+
+  const itemsMock = [
+    {
+      id: 1000,
+      treeIndex: 3,
+      parentId: 2,
+      depth: 2,
+      name: 'Node name',
+      subtreeSize: 4,
+      boundingBox: {
+        max: [0, 0, 0],
+        min: [0, 0, 0],
+      },
+    },
+  ];
+
+  test('list 3d nodes (mocked)', async () => {
+    mock = new MockAdapter(client._instance);
+    const regExp = new RegExp(
```

Comment:
You don't need any regex here? Just a string

Reply:
It's a part of the string, not a full path